### PR TITLE
Set parent api for post, edit and delete using axios

### PR DESCRIPTION
**Pull request by Muh-Baharuddin**
please review my code

### CODE REVIEW

File: src/containers/Supplier/Modal/AddModalSupplier.tsx [hunk 1]
```diff
@@ -18,13 +19,21 @@ function AddModalSupplier(props: handleShowType) {
...
+      toast.error("Maaf terjadi kesalahan pada server. Mohon coba kembali dalam beberapa saat.");
+      handleCloseAdd();
+    })
+    // axios
+    //   .post(`${process.env.API_ENDPOINT}supplier`, data)
+    //   .then(() => {
+    //     toast.success('Data berhasil ditambahkan');
+    //     refreshSupplier();
+    //     handleCloseAdd();
+    //   })
```
Comment (Muh-Baharuddin) [inline src/containers/Supplier/Modal/AddModalSupplier.tsx]: already delete in the next commit

File: services/api.ts [hunk 2]
```diff
@@ -14,3 +14,43 @@ export const get = async <T>(url: string, params: ParamsType): Promise<{
...
+  message: string
+}> => {
+  const deleteUrl = `${url}/${id}`;
+  const response = await axios.delete(deleteUrl);
+
+  return {
+    statusCode: response.status,
+    message: response.data.message,
+  };
+}
```
Comment (sidratul) [inline services/api.ts:56]: dont send id. unless you want to send param

File: services/supplier/index.ts [hunk 3]
```diff
@@ -5,3 +5,15 @@ import { SupplierData } from "./types";
...
+export const postSupplier = (data: SupplierData) => {
+  return api.post<SupplierData>(`${process.env.API_ENDPOINT}supplier`, data);
+}
+
+export const updateSupplier = (id: string, data: SupplierData) => {
+  return api.patch<SupplierData>(`${process.env.API_ENDPOINT}supplier`, id, data);
+}
+
+export const deleteSupplier = (id: string) => {
+  return api.remove(`${process.env.API_ENDPOINT}supplier`, id);
```
Comment (sidratul) [inline services/supplier/index.ts:18]: only send url

File: services/supplier/index.ts [hunk 4]
```diff
@@ -5,3 +5,15 @@ import { SupplierData } from "./types";
 export const getSuppliers = (params: QueryParamsType) => {
   return api.get<SupplierData>(`${process.env.API_ENDPOINT}supplier`, params);
```
Comment (sidratul) [inline services/supplier/index.ts:6]: create `supplier_url=`${process.env.API_ENDPOINT}supplier`

File: services/api.ts [hunk 5]
```diff
@@ -14,3 +14,43 @@ export const get = async <T>(url: string, params: ParamsType): Promise<{
...
+  return {
+    data: response.data,
+    statusCode: response.status,
+  };
+}
+
+export const patch = async <T>(url: string, id: string, data: T): Promise<{
+  data: T,
+  statusCode: number,
+  message: string
```
Comment (sidratul) [inline services/api.ts:33]: create type when duplicate

File: services/api.ts [hunk 6]
```diff
@@ -14,3 +14,43 @@ export const get = async <T>(url: string, params: ParamsType): Promise<{
     statusCode: response.status,
   };
 }
+
+export const post = async <T>(url: string, data: T): Promise<{
+  data: T,
+  statusCode: number,
+}> => {
+  const response = await axios.post(url, data);
```
Comment (sidratul) [inline services/api.ts:22]: write to `const response = await axios.post<T>(url, data);`

File: services/api.ts (hunk 2 above)
Comment (Muh-Baharuddin) [reply services/api.ts]: done only send url

File: services/supplier/index.ts (hunk 3 above)
Comment (Muh-Baharuddin) [reply services/supplier/index.ts]: done

File: services/api.ts (hunk 5 above)
Comment (Muh-Baharuddin) [reply services/api.ts]: done create type

File: services/api.ts (hunk 6 above)
Comment (Muh-Baharuddin) [reply services/api.ts]: done

File: services/api.ts [hunk 7]
```diff
@@ -1,16 +1,36 @@
...
     data: response.data,
     statusCode: response.status,
   };
 }
+
+export const remove = async (url: string): Promise<ResponseType> => {
+  const response = await axios.delete(url);
+  return {
+    statusCode: response.status,
+    message: response.data.message,
```
Comment (sidratul) [inline services/api.ts:34]: fix this. message not always exists

File: services/api.ts [hunk 8]
```diff
@@ -1,16 +1,36 @@
...
-  const response = await axios.get(url, {
+export const get = async <T>(url: string, params: ParamsType): Promise<ResponseDataType<T>> => {
+  const response = await axios.get<T>(url, {
     params,
   });
+  return {
+    data: response.data,
+    statusCode: response.status,
+  };
+}
```
Comment (sidratul) [inline services/api.ts:12]: handle error

File: services/supplier/index.ts [hunk 9]
```diff
@@ -2,6 +2,20 @@ import * as api from "services/api";
...
+
+export const postSupplier = (data: SupplierData) => {
+  return api.post<SupplierData>(supplier_url, data);
+}
+
+export const updateSupplier = (params: string, data: SupplierData) => {
+  return api.patch<SupplierData>(supplier_url + "/" + params, data);
+}
+
+export const deleteSupplier = (params: string) => {
```
Comment (sidratul) [inline services/supplier/index.ts:19]: change name to more specific.

File: services/types.ts [hunk 10]
```diff
@@ -13,4 +14,15 @@ export type QueryParamsType = {
...
 export type ParamsType = {
   [key: string]: number | string | Date,
-}
\ No newline at end of file
+}
+
+
+export type ResponseType = {
+  statusCode: number,
+  message?: string,
```
Comment (sidratul) [inline services/types.ts:22]: no need

File: services/types.ts [hunk 11]
```diff
@@ -13,4 +14,15 @@ export type QueryParamsType = {
 
 export type ParamsType = {
   [key: string]: number | string | Date,
-}
\ No newline at end of file
+}
+
+
+export type ResponseType = {
```
Comment (sidratul) [inline services/types.ts:20]: no need. just combine with data

File: services/supplier/index.ts (hunk 9 above)
Comment (Muh-Baharuddin) [reply services/supplier/index.ts:19]: done change to id

File: services/types.ts (hunk 11 above)
Comment (Muh-Baharuddin) [reply services/types.ts:20]: done

File: services/api.ts (hunk 7 above)
Comment (Muh-Baharuddin) [reply services/api.ts:34]: done

File: services/api.ts (hunk 8 above)
Comment (Muh-Baharuddin) [reply services/api.ts:12]: next time

**Pull request merged**